addons/digest/wizard/digest_test.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, tools
+
+
+class TestDigest(models.TransientModel):
+    _name = 'digest.digest.test'
+    _description = 'Sample Digest Wizard'
+
+    email_to = fields.Text(string='Recipients', required=True,
+        help='Carriage-return-separated list of email addresses.', default=lambda self: self.env.user.email_formatted)
+    digest_id = fields.Many2one('digest.digest', string='Digest', required=True, ondelete='cascade')
+
+    def send_mail_test(self):
+        self.ensure_one()
+
+        valid_emails = []
+        for candidate in self.email_to.splitlines():
+            parts = tools.email_split(candidate)
+            if parts:
+                valid_emails.append(parts[0])
+
+        users = self.env['res.users'].search([
```

Comment:
That's ... strange. If I add a sample email to test the output but without any user attached, it won't work ? Maybe recipients should be directly users then. Or alternatively, if an email is not linked to any user, we could render based on current user but force the to to the given email (with for example a force_email_to parameter to the send method).

Reply:
Thanks, I changed with the first solution (users).